db/write_callback_test.cc

```diff
@@ -184,7 +190,8 @@ TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
                       ASSERT_TRUE(writer->callback->Callback(nullptr).ok() ==
                                   !write_group.back().callback_.should_fail_);
                     }
-
+                    
+                    threads_waiting.fetch_add(1);
```

Comment:
> The patch move the update to threads_waiting to the end of WriteThread::JoinBatchGroup:Wait callback to avoid this happening.

How does this move gives time to the leader to update the state of the followers?